feat: add transfer and fix lamport issue

parthesh28 opened the pull request:
i have added the transfer instruction and tried to fix the lamport precision issue in the demand airdrop command. i hope it works. (i have used the solana-sdk crate to implement a few things, hope it doesn't add a lot of dependencies). the command works on devnet. i hope it works
<img width="902" height="732" alt="image" src="https://github.com/user-attachments/assets/629d597e-c5cf-4447-ac2e-4c2bced87d1a" />

BretasArthur1 commented:
Hey @parthesh28 thanks for your PR! Currently we have #17 implementing the transfer feature... About the issue on lamports airdrop you can open a PR for it, but I'm not seeing the bug...

Nagaprasadvr commented:
@parthesh28 ty for the contribution, but this is a duplicate but feel free to open a new pr for airdrop

parthesh28 commented:
> Hey @parthesh28 thanks for your PR! Currently we have #17 implementing the transfer feature... About the issue on lamports airdrop you can open a PR for it, but I'm not seeing the bug...

hey, thank you for reviewing my work! was more than happy to contribute! i forgot to check if it was already done or not and just started working on the feature. regarding the lamport issue, it is a very small change. if you look closely, the request_sol_airdrop function was considering the value given by the user lamports just like the helper function and not sol, so i just added a small fix, where the value given by the user is considered as sol value and converted to lamports to gain proper airdrop!
it is a very small fix, if you believe, i can open a PR for that! let me know!
<img width="1162" height="61" alt="image" src="https://github.com/user-attachments/assets/4a07fdbe-cd0b-417a-ad4e-36b54db2617e" />

BretasArthur1 commented:
Sure! Open a PR
